### PR TITLE
Feature/gcb 355 configurable title for photos widget

### PR DESCRIPTION
**Pull request by ncreated**
This PR changes image widget configuration json. `isHeaderVisible ` field is no longer expected. Instead, `header` object must be provided with at least `title` value to have the header rendered:

```
{
  "name": "image",
  "settings": {
    "urls": [],
    "header" : {
      "title" : "Widget title",
      "subtitle" : "Widget subtitle"
    }
  },
}, 
```

- `header` is optional (if missing, no header is added);
- both `header` and `title` are required if the header should be added;
- `subtitle`  is optional;

--

Make sure these boxes are checked before submitting your Pull Request - thank you!

- [x] [SwiftLint](https://github.com/Realm/SwiftLint) is installed on your system
- [x] The project does not generate any warnings or errors
- [x] The tests are passing
- [x] Branch name is in line with [GitFlow](http://jeffkreeftmeijer.com/2010/why-arent-you-using-git-flow/)

### CODE REVIEW

File: GrandCentralBoard/Widgets/Image/ImageWidgetBuilder.swift [hunk 1]
```diff
@@ -20,11 +20,10 @@ final class ImageWidgetBuilder: WidgetBuilding {
     func build(settings: AnyObject) throws -> WidgetControlling {
         let settings = try ImageWidgetConfiguration.decode(settings)
 
-        let imagesSource = try RemoteImageSource(paths: settings.imagePaths, dataDownloader: dataDownloader)
         let view = ImageWidgetView.fromNib()
+        let imagesSource = try RemoteImageSource(paths: settings.imagePaths, dataDownloader: dataDownloader)
+        let headerConfiguration = settings.header.flatMap { (title: $0.title, subtitle: $0.subtitle ?? "") }
```
Comment (michallaskowski) [inline GrandCentralBoard/Widgets/Image/ImageWidgetBuilder.swift]: You can use `map` here (just saying :)

File: GrandCentralBoardTests/Widgets/Image/ImageWidgetSnapshotTests.swift [hunk 2]
```diff
@@ -42,21 +42,45 @@ final class ImageWidgetSnapshotTests: FBSnapshotTestCase {
...
+        let source = try! RemoteImageSource(paths: ["http://macoscope.com"], dataDownloader: dataDownloader)
+        let header: ImageWidgetHeader? = nil
+
+        let widget = ImageWidget(view: widgetView, sources: [source], header: header)
+        widget.update(source)
+
+        FBSnapshotVerifyView(widget.view)
+    }
+
+    func testViewWithCatImageAndCustomHeader() {
```
Comment (michallaskowski) [inline GrandCentralBoardTests/Widgets/Image/ImageWidgetSnapshotTests.swift]: Can you add one test with title, but no subtitle?

File: GrandCentralBoard/Widgets/Image/ImageWidget.swift [hunk 3]
```diff
@@ -7,28 +7,31 @@ import UIKit
 import GCBCore
 
 
+typealias ImageWidgetHeader = (title: String, subtitle: String)
```
Comment (michallaskowski) [inline GrandCentralBoard/Widgets/Image/ImageWidget.swift]: I am not sure what fits better here, a struct (like we used to) or a tuple.

**Comment by michallaskowski** [comment]
That's all, minor things, good job 👍

**Comment by ncreated** [comment]
@michallaskowski all done 🎾

**Pull request merged**